enhancements/cluster-runtime-constraints.md

```diff
@@ -0,0 +1,138 @@
+---
+title: Cluster Runtime Constraint
+authors:
+  - "@dinhxuanvu"
+reviewers:
+  - "@kevinrizza"
+approvers:
+  - "@kevinrizza"
+creation-date: 2021-08-26
+last-updated: 2020-09-06
+status: provisional
+---
+
+# cluster-runtime-constraint
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Summary
+
+This proposal presents the specification and usage to enable runtime constraints to be considered during operator dependency resolution. This feature will ensure Operator Lifecycle Manage (OLM) to install operators that satisfy dependency requirements and runtime constraints if present.
```

Comment:
For an outsider like me, It seems like the abstract is really something like this:

Give Cluster Admins and Operator Developers the ability to restrict provisioning of an Operator Bundle based on the Bundle Properties.